egs/wham/ConvTasNet/run.sh

```diff
@@ -61,17 +61,27 @@ train_dir=$dumpdir/tr
 valid_dir=$dumpdir/cv
 test_dir=$dumpdir/tt
 
-if [[ $stage -le  0 ]]; then
-  echo "Stage 0: Converting sphere files to wav files"
-  . local/convert_sphere2wav.sh --sphere_dir $sphere_dir --wav_dir $wsj0_wav_dir
+# Generate a random ID for the run if no tag is specified
+uuid=$(python3 -c 'import uuid, sys; print(str(uuid.uuid4())[:8])')
+if [[ -z ${tag} ]]; then
+	tag=${task}_${uuid}
+fi
+expdir=exp/train_convtasnet_${tag}
+mkdir -p $expdir && echo $uuid >> $expdir/run_uuid.txt
+echo "Results from the following experiment will be stored in $expdir"
+
+
+if [ $stage -le 0 ]; then
+    echo "Stage 0: Converting sphere files to wav files"
+    local/convert_sphere2wav.sh --sphere_dir $sphere_dir --wav_dir $wsj0_wav_dir
 fi
 
-if [[ $stage -le  1 ]]; then
-	echo "Stage 1: Generating 8k and 16k WHAM dataset"
-  . local/prepare_data.sh --wav_dir $wsj0_wav_dir --out_dir $wham_wav_dir --python_path $python_path
+if [ $stage -le 1 ]; then
+    echo "Stage 1: Generating 8k and 16k WHAM dataset"
+    local/prepare_data.sh --wav-dir $wsj0_wav_dir --out-dir $wham_wav_dir
```

Comment:
Why remove the `.` ?

Reply:
because we are not supposed to have directory jumping in `run.sh` so the dot is not necessary. And we do not have other `local` folders.

In some good bash practice from my point of view (like kaldi), we avoid directory jumping at very high level. I do not know if in sub scripts from folders such jumping via `cd` and `cd -` is good practice. Of course, if they are wrapped in functions, they are for sure.

Reply:
> In some good bash practice from my point of view (like kaldi), we avoid directory jumping at very high level.

I agree with that.